libraries/chain/db_maint.cpp

```diff
@@ -925,7 +925,29 @@ void database::perform_chain_maintenance(const signed_block& next_block, const g
    // Reset all BitAsset force settlement volumes to zero
    for( const auto& d : get_index_type<asset_bitasset_data_index>().indices() )
    {
-      modify( d, [](asset_bitasset_data_object& o) { o.force_settled_volume = 0; });
+      if(head_block_time() < HARDFORK_CORE_518_TIME)
+         modify( d, [](asset_bitasset_data_object& o) { o.force_settled_volume = 0; });
+      else { // clean expired feeds
+         modify( d, [this](asset_bitasset_data_object& o) {
+            o.force_settled_volume = 0;
+
+            // Check if asset is smartcoin
+            const auto& settlement_price = o.feeds.begin()->second.second.settlement_price;
+            if(!settlement_price.is_null()) {
```

Comment:
This check is incorrect. The first feed is null doesn't mean all feeds are null.

Reply:
https://github.com/bitshares/bitshares-core/pull/832#issuecomment-384952153
you think that is wrong ? makes sense to me.

Reply:
waiting for more comments from @pmconrad maybe.